tuna/miopen/subcmd/load_job.py

```diff
@@ -40,87 +42,21 @@
-
-
-def parse_args():
-  """ Argument input for the module """
-  #pylint: disable=duplicate-code
-  parser = setup_arg_parser(
-      'Insert jobs into MySQL db by tag from" \
-      " config_tags table.', [TunaArgs.VERSION, TunaArgs.CONFIG_TYPE])
-  config_filter = parser.add_mutually_exclusive_group(required=True)
-  solver_filter = parser.add_mutually_exclusive_group()
-  config_filter.add_argument(
-      '-t',
-      '--tag',
-      type=str,
-      dest='tag',
-      default=None,
-      help='All configs with this tag will be added to the job table. \
-                        By default adds jobs with no solver specified (all solvers).'
-  )
-  config_filter.add_argument('--all_configs',
-                             dest='all_configs',
-                             action='store_true',
-                             help='Add all convolution jobs')
-  solver_filter.add_argument(
-      '-A',
-      '--algo',
-      type=str,
-      dest='algo',
-      default=None,
-      help='Add job for each applicable solver+config in Algorithm.',
-      choices=ALG_SLV_MAP.keys())
-  solver_filter.add_argument('-s',
-                             '--solvers',
-                             type=str,
-                             dest='solvers',
-                             default=None,
-                             help='add jobs with only these solvers '\
-                               '(can be a comma separated list)')
-  parser.add_argument(
-      '-d',
-      '--only_dynamic',
-      dest='only_dynamic',
-      action='store_true',
-      help='Use with --tag to create a job for dynamic solvers only.')
-  parser.add_argument('--tunable',
-                      dest='tunable',
-                      action='store_true',
-                      help='Use to add only tunable solvers.')
-  parser.add_argument('-c',
-                      '--cmd',
-                      type=str,
-                      dest='cmd',
-                      default=None,
-                      required=False,
-                      help='Command precision for config',
-                      choices=['conv', 'convfp16', 'convbfp16'])
-  parser.add_argument('-l',
-                      '--label',
-                      type=str,
-                      dest='label',
-                      required=True,
-                      help='Label to annontate the jobs.',
-                      default='new')
-  parser.add_argument('--fin_steps', dest='fin_steps', type=str, default=None)
-  parser.add_argument(
-      '--session_id',
-      required=True,
-      dest='session_id',
-      type=int,
-      help=
-      'Session ID to be used as tuning tracker. Allows to correlate DB results to tuning sessions'
-  )
-
-  args = parser.parse_args()
 
+def arg_fin_steps(args: argparse.Namespace):
+  """fin steps for load jobs"""
   if args.fin_steps:
     steps = [x.strip() for x in args.fin_steps.split(',')]
     args.fin_steps = set(steps)
 
+
+def arg_solvers(
```

Comment:
this function is not called, this causes a bug in that your PR does not take into account when -A or -s is passed to load_job. 
For instance, if you run this cmd in develop:
```
./tuna/miopen/subcmd/load_job.py -t recurrent_rk_prase_jobs_61 -l recurrent_rk_prase_jobs_61 --session_id 1 -A miopenConvolutionAlgoGEMM
```
this is what args looks like:
```
./tuna/miopen/subcmd/load_job.py -t recurrent_rk_prase_jobs_61 -l recurrent_rk_prase_jobs_61 --session_id 1 -A miopenConvolutionAlgoGEMM
Namespace(algo='miopenConvolutionAlgoGEMM', all_configs=False, cmd=None, config_type=<ConfigType.convolution: 'convolution'>, fin_steps=None, label='recurrent_rk_prase_jobs_61', only_dynamic=False, session_id=1, solvers=[('GemmFwd1x1_0_1', 88), ('GemmFwd1x1_0_1_int8', 89), ('GemmFwd1x1_0_2', 90), ('GemmFwdRest', 91), ('GemmBwd1x1_stride2', 95), ('GemmBwd1x1_stride1', 96), ('GemmBwdRest', 97), ('GemmWrw1x1_stride1', 101), ('GemmWrwUniversal', 102)], tag='recurrent_rk_prase_jobs_61', tunable=False, yaml=None)
```
As you can see it picked up the solvers for that -A algo.
Now, if you run the same command in your PR, this is what args looks like:
```
 ./tuna/go_fish.py miopen load_job -t recurrent_rk_prase_jobs_61 -l recurrent_rk_prase_jobs_61 --session_id 1 -A miopenConvolutionAlgoGEMM
2023-04-04 14:33:16,997 - go_fish - INFO -  [go_fish.py:71] - ['./tuna/go_fish.py', 'miopen', 'load_job', '-t', 'recurrent_rk_prase_jobs_61', '-l', 'recurrent_rk_prase_jobs_61', '--session_id', '1', '-A', 'miopenConvolutionAlgoGEMM']
2023-04-04 14:33:16,997 - go_fish - INFO -  [go_fish.py:72] - 11
Namespace(algo='miopenConvolutionAlgoGEMM', all_configs=False, cmd=None, config_type=<ConfigType.convolution: 'convolution'>, fin_steps=None, label='recurrent_rk_prase_jobs_61', only_dynamic=False, session_id=1, solvers=None, tag='recurrent_rk_prase_jobs_61', tunable=False, yaml=None)

```
As you can see, the args results from this run do not contain any solvers, that is because the arg_solver function is never called. 
Please address

Reply:
@reidkwja  given that this was indeed the issue that was causing bad json file and hence an miopen segfault, please update the load_job unit test and look for:

1. correct number of jobs being set for the expected solvers in a given algo, specified with -A
2. correct solver being set for a job that specifies a solver